internal/util/type_conversion_test.go

```diff
@@ -0,0 +1,28 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package util
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestToOtelValue(t *testing.T) {
+	testCases := []struct {
```

Comment:
Do you have time to add test cases for all the other types, while you're here...